src/common/components/BottomNavigation/index.jsx

```diff
@@ -1,20 +1,28 @@
 import React, { useState, useEffect } from "react";
 import { View, Pressable } from "react-native";
 import { useDispatch, useSelector } from "react-redux";
+import { useNavigation } from "@react-navigation/native";
 import PropTypes from "prop-types";
 import useTheme from "@common/context";
 import useThemedStyles from "@common/hooks/useThemedStyles";
 import { HomeIcon, SettingsIcon, MusicIcon, ReadIcon } from "@common/icons";
 import { CustomText, actions, constant, STRINGS, SafeArea } from "@common";
 import createStyles from "./style";
 
-const BottomNavigation = ({ navigation, activeKey }) => {
+const BottomNavigation = ({ activeKey }) => {
+  const navigation = useNavigation();
   const dispatch = useDispatch();
   const { theme } = useTheme();
   const styles = useThemedStyles(createStyles);
   const isAudio = useSelector((state) => state.isAudio);
-  const currentBani = useSelector((state) => state.currentBani);
   const [isSettings, setIsSettings] = useState(false);
+  const [previousRouteName, setPreviousRouteName] = useState(null);
+
+  // Helper function to get current route name
+  const getCurrentRouteName = () => {
+    const navState = navigation.getState();
+    return navState?.routes[navState?.index]?.name;
+  };
```

Comment:
The `getCurrentRouteName` function is recreated on every render. Consider memoizing it or using `useCallback` if it's called frequently.